geomstats/geometry/product_manifold.py

```diff
@@ -14,101 +14,288 @@
-    [..., dim_1 + ... + dim_n_manifolds]
-    where n_manifolds is the number of manifolds in the product.
-    This type of representation is called 'vector'.
-
-    Alternatively, a point can be represented by an array of shape:
-    [..., n_manifolds, dim] if the n_manifolds have same dimension dim.
-    This type of representation is called `matrix`.
-
     Parameters
     ----------
-    manifolds : list
+    factors : list
         List of manifolds in the product.
+    metrics : list
+        List of metrics for each manifold
+        If a list of positive numbers is given, the metric is obtained from the manifold
+        and scaled by the number.
+        Optional, default value obtained from metric property of manifold
+    default_point_type : {'vector', 'matrix}
+        Optional. Vector representation gives the point as a 1-d array.
+        Matrix representation allows for a point to be represented by an array of shape
+        (n, dim), if each manifold has default_point_type 'vector' with shape (dim,).
     n_jobs : int
         Number of jobs for parallel computing.
         Optional, default: 1.
     """
 
-    # FIXME (nguigs): This only works for 1d points
-
-    def __init__(
-        self, manifolds, metrics=None, default_point_type="vector", n_jobs=1, **kwargs
-    ):
+    def __init__(self, factors, metrics=None, default_point_type="vector", **kwargs):
         geomstats.errors.check_parameter_accepted_values(
             default_point_type, "default_point_type", ["vector", "matrix"]
         )
 
-        self.dims = [manifold.dim for manifold in manifolds]
+        self.factors = factors
+
+        dim = sum(self.factor_dims)
+
+        if default_point_type == "vector":
+            shape = (sum([prod(factor_shape) for factor_shape in self.factor_shapes]),)
+        else:
+            if (self.factor_shapes.count(self.factor_shapes[0]) ==
+                    len(self.factor_shapes)):
+                if len(self.factor_shapes[0]) == 1:
+                    shape = (len(self.factors), *self.factors[0].shape)
+                else:
+                    raise ValueError(
+                        "A default_point_type of \'matrix\' can only be used if all "
+                        "manifolds have vector type."
+                    )
+            else:
+                raise ValueError(
+                    "A default_point_type of \'matrix\' can only be used if all "
+                    "manifolds have the same shape."
+                )
+
+        if "extrinsic" in self.factor_default_coords_types:
+            default_coords_type = "extrinsic"
+        else:
+            default_coords_type = "intrinsic"
+
         if metrics is None:
-            metrics = [manifold.metric for manifold in manifolds]
+            metric_scales = None
+            metrics = [manifold.metric for manifold in factors]
+        elif gs.all([is_positive(metric) for metric in metrics]):
+            metric_scales = metrics
```

Comment:
call it `metrics_scales` in argument. Very confusing otherwise. The fact we will not have `metrics` there anymore makes things more clear.